mantidproject/mantid
# Draw shape around entire tube when using pick tool on ALFView

**Description of work.**
This PR makes it possible to use the `Select whole tube` toolbar option to select an entire tube. Whenever a tube is selected, a rectangle is now drawn around the entirety of the tube, which provides a far better visual indication for which tubes are selected.

Each tube has its own separate rectangle drawn around it, and this makes it easier to delete a region on a specific tube but keep a region around another tube.

**To test:**
1. Open ALFView and load ALF82301
2. Click on Pick tab
3. Expand Rebin section. Rebin with 5.5,0.01,6
4. Click the 'Draw a rectangle' tool at the top of the Pick tab
5. Select the two tubes in the centre with the yellow peak.
6. Two separate rectangles should be drawn around each of the tubes automatically.
7. Click the 'Select whole tube' tool at the top of the Pick tab
8. Click on individual tubes to add them to the selection. The right hand size plot should update

Fixes #34835

*This does not require release notes* because **release notes will be added in a separate PR**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
